wrapper/src/main/java/software/amazon/jdbc/wrapper/ConnectionWrapper.java

```diff
@@ -139,15 +146,34 @@ protected void init(
     }
   }
 
-  protected String getProtocol(final String url) {
+  protected DatabaseDialect getDialect(final String url, Properties info) {
+    String dialectClassName = DATABASE_DIALECT.getString(info);
```

Comment:
One of the topic that has been discussed a long time ago was to avoid providing class names in a connection string. Configuration Profiles is a feature that allows load custom plugins for enterprise users and to avoid class names in a connection string. I'd suggest to consider using similar approach for dialect classes.